lib/galaxy/model/__init__.py

```diff
@@ -204,7 +206,21 @@
 
 _datatypes_registry = None
 
-mapper_registry = registry()
+STR_TO_STR_DICT = Dict[str, str]
+
+
+class TransformAction(TypedDict):
+    action: str
+
+
+TRANSFORM_ACTIONS = List[TransformAction]
+
+mapper_registry = registry(
+    type_annotation_map={
+        Optional[STR_TO_STR_DICT]: JSONType,
+        Optional[TRANSFORM_ACTIONS]: MutableJSONType,
```

Comment:
```suggestion
        STR_TO_STR_DICT: JSONType,
        TRANSFORM_ACTIONS: MutableJSONType,
```
We use `Optional` in the model definition

Reply:
Well my thinking was the thing in map needed to match the column definition completely. If it doesn't have to match the column definition exactly... I think we don't actually need this map right? I am thinking we just drop this if it isn't actually needed. We would have test failures if this was needed and wrong.

Reply:
@jmchilton it works as intended; I've reread the docs and tested it to make sure I'm not misreading stuff. Here's how I read it:
```py
STR_TO_STR_DICT = Dict[str, str]  # we define a type alias
...
STR_TO_STR_DICT: JSONType  
# in the statement above (type annotation map), we say that the Python type STR_TO_STR_DICT 
# should be mapped to SQLAlchemy's JSONType, which is a SQLAlchemy type that knows how to 
# generate its DDL. 
# If we substituted JSONType for Integer, the Python code would still expect Dict[str, str],
# but in the db that field would be defined as an integer type.
...
object_store_id_overrides: Mapped[Optional[STR_TO_STR_DICT]]  # that's all we need here!
# the field will be typed as Optional[Dict[str, str]], represented as bytea in the database. 
```
The `Optional` type specifier works fine: here's an example that causes a mypy error; removing `Optional` fixes it - as expected:
```py
def test(job: Job):
    job.object_store_id_overrides['key']
test(Job())

# lib/galaxy/model/__init__.py:11430: error: Value of type "Optional[dict[str, str]]" is not indexable  [index]
```

Does this make sense?

(There's a conflict to resolve - I'm happy to take care of this.)

Reply:
Maybe this makes sense - I think given that we don't have a typed way to distinguish between what should be mutable or not currently we should drop the dictionary and keep being explicit but I will defer to you until I understand it better.

Reply:
Yes, i think so (re. your update). Also, that type annotation map may be useful for other type mappings (SA's docs have some examples [here](https://docs.sqlalchemy.org/en/20/changelog/whatsnew_20.html#step-five-make-use-of-pep-593-annotated-to-package-common-directives-into-types)), even if the only benefit in those cases would be more succinct code.